### PR TITLE
Entra ID sync enhancements commit July 19, 2024

### PR DESCRIPTION
**Pull request by Dynamicspedia**
Entra ID group synchronization enhancements. Features tested and working. There are a few BP warnings left. Labels added, but not translated to other languages currently.

### CODE REVIEW

**Comment by Dynamicspedia** [comment]
It looks like my commit was not synced with GitHub. I will check what is the status and why my added objects are not visible now.

**Comment by Dynamicspedia** [comment]
it looks like the actual F&O objects are now available in this pull request.

File: Project/D365FOAdminToolkit/D365FOAdminToolkit.sln
Comment (jofme) [inline Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: This looks wrong...

Comment (Dynamicspedia) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: Still something wrong? If so, please let me know. I'm not a GIT expert (yet).

Comment (jofme) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: I was just wondering why the solution and project files was deleted 😅

Comment (jofme) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: Should be fairly easy to fix if you make another check in of the files in the branch

Comment (Dynamicspedia) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: I tried to exclude the changes as the project files are not compatible between a OneBox and the new UDE development environments. When required, I can add them back.

Comment (Dynamicspedia) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: When I now look at the status, the checks are not successful. The step "Build 10.0.38 (pull request)" failed. Not sure what is causing this and how to fix it.

Comment (jofme) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: > When I now look at the status, the checks are not successful. The step "Build 10.0.38 (pull request)" failed. Not sure what is causing this and how to fix it.

It's complaining about a missing EDT

`:\temp\fscps.tools\_bld\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.7.0.7120.53\DevAlm\Microsoft.Dynamics.Framework.Tools.BuildTasks.targets(85,5): error : Error: dynamics://Table/ADMEntraIdGroupUserLastSync(1,14) - The underlying type 'SysUserLastSyncTimeUTC ' or its base type for table 'ADMEntraIdGroupUserLastSync' field 'LastSyncDateTime' does not exist. [c:\temp\fscps.tools\_bld\10.0.38_build\Build\D365FOAdminToolkit.rnrproj]
c:\temp\fscps.tools\_bld\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.7.0.7120.53\DevAlm\Microsoft.Dynamics.Framework.Tools.BuildTasks.targets(85,5): error : Error: AxTable/ADMEntraIdGroupUserLastSync/Fields/LastSyncDateTime/ExtendedDataType - Extended data type 'SysUserLastSyncTimeUTC' does not exist. [c:\temp\fscps.tools\_bld\10.0.38_build\Build\D365FOAdminToolkit.rnrproj]
c:\temp\fscps.tools\_bld\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.7.0.7120.53\DevAlm\Microsoft.Dynamics.Framework.Tools.BuildTasks.targets(85,5): error : Error: dynamics://Table/ADMEntraIdGroupUserLastSync/Method/syncRequired(23,69) - The name 'SysUserLastSyncTimeUTC' does not denote a class, a table, or an extended data type. [c:\temp\fscps.tools\_bld\10.0.38_build\Build\D365FOAdminToolkit.rnrproj]
c:\temp\fscps.tools\_bld\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.7.0.7120.53\DevAlm\Microsoft.Dynamics.Framework.Tools.BuildTasks.targets(85,5): error : Error: dynamics://Table/ADMEntraIdGroupUserLastSync/Method/setLastSync(48,9) - The name 'SysUserLastSyncTimeUTC' does not denote a class, a table, or an extended data type. [c:\temp\fscps.tools\_bld\10.0.38_build\Build\D365FOAdminToolkit.rnrproj]`

It's most likely missing in 10.0.38.
Check-in a small change to the "default" variable in .`github/workflows/build.yml` to 10.0.39 or 10.0.40, then it should work.

```
on:
  pull_request:
  workflow_dispatch:
    inputs:
      version:
        description: 'Select FSCM version. Use wildcard * to let settings .json files decide.'
        required: false
        default: '10.0.40'
```


I did it here for Finance Toolbox: https://github.com/jofme/D365FinanceToolbox/commit/384a85ecf337969ffdd3cffaec076ef5ca255f97

Comment (Dynamicspedia) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: Thank you for your help, Jonas. I have added a new EDT to ensure backwards compatibility. All checks have now passed.

Comment (jofme) [reply Project/D365FOAdminToolkit/D365FOAdminToolkit.sln:1]: > Thank you for your help, Jonas. I have added a new EDT to ensure backwards compatibility. All checks have now passed.

No problem 😊

**Pull request merged**